platform/linux-generic/odp_crypto.c

```diff
@@ -112,8 +115,22 @@ struct odp_crypto_generic_session_t {
 		const EVP_MD *evp_md;
 		crypto_func_t func;
 	} auth;
+
+	/* These bitfields are cleared at odp_crypto_session_destroy()
+	 * together with the rest of data */
+	crypto_valid cipher_valid[(ODP_THREAD_COUNT_MAX + CV_SIZE - 1) /
+				CV_SIZE];
+	crypto_valid hmac_valid[(ODP_THREAD_COUNT_MAX + CV_SIZE - 1) /
+				CV_SIZE];
+	unsigned idx;
 };
 
+#define IS_VALID(session, type, id)	\
+	(session->type ## _valid[id / CV_SIZE] & (1 << (id % CV_SIZE)))
+
+#define SET_VALID(session, type, id)	\
+	session->type ## _valid[id / CV_SIZE] |= (1 << (id % CV_SIZE))
```

Comment:
Checkpatch wants this enclosed in parentheses. Since `IS_VALID` follows that model, no harm in doing so with `SET_VALID` as well.
```
ERROR: Macros with complex values should be enclosed in parentheses
#63: FILE: platform/linux-generic/odp_crypto.c:131:
+#define SET_VALID(session, type, id)	\
+	session->type ## _valid[id / CV_SIZE] |= (1 << (id % CV_SIZE))
```